projects/packages/jetpack-mu-wpcom/src/features/custom-css/custom-css.php

```diff
@@ -0,0 +1,1247 @@
+		 * Set up the actions and filters needed for our compatability layer on top of core's Custom CSS implementation.
+		 */
+		public static function add_hooks() {
+			add_action( 'init', array( __CLASS__, 'init' ) );
+			add_action( 'customize_controls_enqueue_scripts', array( __CLASS__, 'customize_controls_enqueue_scripts' ) );
+			add_action( 'customize_register', array( __CLASS__, 'customize_register' ) );
+			add_filter( 'map_meta_cap', array( __CLASS__, 'map_meta_cap' ), 20, 2 );
+			add_action( 'customize_preview_init', array( __CLASS__, 'customize_preview_init' ) );
+			add_filter( '_wp_post_revision_fields', array( __CLASS__, 'wp_post_revision_fields' ), 10, 2 );
+			add_action( 'load-revision.php', array( __CLASS__, 'load_revision_php' ) );
+
+			add_action( 'wp_enqueue_scripts', array( __CLASS__, 'wp_enqueue_scripts' ) );
+
+			// Handle Sass/LESS.
+			add_filter( 'customize_value_custom_css', array( __CLASS__, 'customize_value_custom_css' ), 10, 2 );
+			add_filter( 'customize_update_custom_css_post_content_args', array( __CLASS__, 'customize_update_custom_css_post_content_args' ), 10, 3 );
+			add_filter( 'update_custom_css_data', array( __CLASS__, 'update_custom_css_data' ), 10, 2 );
+
+			// Stuff for stripping out the theme's default stylesheet...
+			add_filter( 'stylesheet_uri', array( __CLASS__, 'style_filter' ) );
+			add_filter( 'safecss_skip_stylesheet', array( __CLASS__, 'preview_skip_stylesheet' ) );
+
+			// Stuff for overriding content width...
+			add_action( 'customize_preview_init', array( __CLASS__, 'preview_content_width' ) );
+			add_filter( 'jetpack_content_width', array( __CLASS__, 'jetpack_content_width' ) );
+			add_filter( 'editor_max_image_size', array( __CLASS__, 'editor_max_image_size' ), 10, 3 );
+			add_action( 'template_redirect', array( __CLASS__, 'set_content_width' ) );
+			add_action( 'admin_init', array( __CLASS__, 'set_content_width' ) );
+			add_action( 'jetpack_modules_loaded', array( __CLASS__, 'custom_css_loaded' ) );
+		}
+
+		/**
+		 * Things that we do on init.
+		 */
+		public static function init() {
+
+			wp_register_style( 'jetpack-codemirror', plugins_url( 'custom-css/css/codemirror.css', __FILE__ ), array(), '20120905' );
+			wp_register_style( 'jetpack-customizer-css', plugins_url( 'custom-css/css/customizer-control.css', __FILE__ ), array(), '20140728' );
+			wp_register_script( 'jetpack-codemirror', plugins_url( 'custom-css/js/codemirror.min.js', __FILE__ ), array(), '3.16', true );
+
+			$src = Assets::get_file_url_for_environment(
+				'_inc/build/custom-css/custom-css/js/core-customizer-css.core-4.9.min.js',
+				'modules/custom-css/custom-css/js/core-customizer-css.core-4.9.js'
+			);
+			wp_register_script(
+				'jetpack-customizer-css',
+				$src,
+				array(
+					'jquery',
+					'customize-controls',
+					'underscore',
+				),
+				JETPACK__VERSION,
+				true
+			);
+
+			wp_register_script(
+				'jetpack-customizer-css-preview',
+				Assets::get_file_url_for_environment(
```

Comment:
As with the other similar comment, it might be worth rewriting this in relation to `get_file_url_for_environment`.

Reply:
Hmm that other comment didn't get added - re-added now: https://github.com/Automattic/jetpack/pull/37794/files#r1637844413